x/validator-preference/keeper/msg_server.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/v12/x/validator-preference/types"
+)
+
+type msgServer struct {
+	keeper *Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
+	return &msgServer{
+		keeper: keeper,
+	}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) SetValidatorSetPreference(goCtx context.Context, msg *types.MsgSetValidatorSetPreference) (*types.MsgSetValidatorSetPreferenceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	preferences := msg.Preferences
+
+	// check if a user already have a validator-set created
+	existingValidator, found := server.keeper.GetValidatorSetPreference(ctx, msg.Delegator)
+	if found {
+		preferences = existingValidator.Preferences
```

Comment:
hrmmm I think we should be either
1. return err and stop execution when preference is same as the previous preferences
2. overwrite existing preferences to the new preference

Reply:
1. Resolved
2. I realized that i'm already using `SetValidatorSetPreferences` that can handle overrides. If its a new val-set it'll register the storeKey(userAddr), if its existing val-set it'll use the storeKey(userAddr) and new preferences

Reply:
lmk what you think :))